### DISCUSSION TITLE
Should we steal focus from other windows upon activating the global hotkey?

### DISCUSSION THREAD

**Discussion by guihkx** [nuttyartist/notes#477] [Ideas]
I'm working on [this rather controversial change](https://github.com/nuttyartist/notes/compare/dev...guihkx:notes:steal-focus) (build artifacts for [linux](https://github.com/guihkx/notes/actions/runs/4193045418), [mac](https://github.com/guihkx/notes/actions/runs/4193045430), [windows](https://github.com/guihkx/notes/actions/runs/4193045417)), which allows us to steal focus from other windows when the user hits the global hotkey to summon or hide Notes.

On Linux (tested on GNOME 43 / KDE Plasma 5.26), if Notes is not stacked at the very top of the windows list, or also when it's minimized, hitting the summon hotkey will yield this behavior:

https://user-images.githubusercontent.com/626206/219337994-52797576-9f8d-4175-981d-f5bbde12e1bd.mp4

<br>
Which I believe is done on purpose by window managers to control bad-behaving apps that steal focus...<br><br>

However, we could use this ~~excuse~~ argument to bypass that: Because users purposely pressed our global hotkey, it means that they need Notes to steal focus from other windows *right now*! ... Right? 😬 

Windows (tested on 7 and 11) seems to have a relaxed rule about this: It will always steal focus from other windows as long as we're not minimized.

macOS was all weird on my tests, so I added two hacks there.

Please share your thoughts.

**Comment by bjorn** [comment]
> However, we could use this ~excuse~ argument to bypass that: Because users purposely pressed our global hotkey, it means that they need Notes to steal focus from other windows _right now_! ... Right? :grimacing:

Well, I agree with this argument.

**Reply by nuttyartist** [reply]
Haha me too!

**Comment by guihkx** [comment]
Alright then 😄 

I'll send a PR for review!
